src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy

```diff
@@ -0,0 +1,65 @@
+package com.causecode.fileuploader
+
+import grails.test.mixin.TestFor
+import org.springframework.web.multipart.MultipartFile
+import spock.lang.Specification
+
+/**
+ * Unit tests for {@link UtilitiesService}
+ */
+@TestFor(UtilitiesService)
+class UtilitiesServiceSpec extends Specification {
+
+    void "test getProviderInstance method"() {
+        when: 'getProviderInstance method is called and class does not exist'
+        service.getProviderInstance('test')
+
+        then: 'Method should throw exception'
+        ProviderNotFoundException e = thrown()
+        e.message == 'Provider test not found.'
+
+        when: 'getProviderInstance method is called and class exist'
+        service.getProviderInstance('Amazon')
+
+        then: 'No exception is thrown'
+        noExceptionThrown()
+    }
+
+    void "test getNewTemporaryDirectoryPath method"() {
```

Comment:
Please complete the sentences. It's really not clear what are you trying to test here. Whether it's a success case or failure and what are the expectations.

Reply:
Done.